kubectl-cloudflow/cmd/verify.go

```diff
@@ -0,0 +1,170 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/go-akka/configuration"
+	"github.com/lightbend/cloudflow/kubectl-cloudflow/domain"
+	"github.com/lightbend/cloudflow/kubectl-cloudflow/util"
+	"github.com/lightbend/cloudflow/kubectl-cloudflow/verify"
+	"github.com/spf13/cobra"
+)
+
+type verifyBlueprintCMD struct {
+	cmd *cobra.Command
+}
+
+func init() {
+	verifyBlueprintCMD := &verifyBlueprintCMD{}
+	verifyBlueprintCMD.cmd = &cobra.Command{
+		Use:     "verify",
+		Short:   "Verifies the blueprint of a Cloudflow Application.",
+		Example: "kubectl cloudflow verify <blueprint-file-name>",
+		Run:     verifyBlueprintCMD.verifyImpl,
+		Args:    validateVerifyCmdArgs,
+	}
+	rootCmd.AddCommand(verifyBlueprintCMD.cmd)
+}
+
+// this map is constructed entirely from blueprint
+func getImageRefsFromConfig(config *configuration.Config) map[string]domain.ImageReference {
+	// get the images from the blueprint
+	images := config.GetNode("blueprint.images").GetObject().Items()
+
+	var imageKeyVals = make(map[string]domain.ImageReference)
+
+	for imageKey, imageRef := range images {
+		ref, err := verify.ParseImageReference(imageRef.GetString())
+		if err != nil {
+			util.LogAndExit(err.Error())
+		}
+		imageKeyVals[imageKey] = *ref
+	}
+	return imageKeyVals
+}
+
+func getFilePath(pathToBlueprint string) (string, error) {
+	blueprintURL, err := url.Parse(pathToBlueprint)
```

Comment:
This function looks more general than being specific to blueprint loading. Can we move it to some place else (maybe `domain`) so that it can be reused by others as well, if need be ?

Reply:
it is possible.